langchain/vectorstores/weaviate.py

```diff
@@ -212,6 +212,7 @@ def from_texts(
                 )
         """
         weaviate_url = get_from_dict_or_env(kwargs, "weaviate_url", "WEAVIATE_URL")
+        weaviate_api_key = get_from_dict_or_env(kwargs, "weaviate_api_key", "WEAVIATE_API_KEY")
```

Comment:
i think this requires that the key is required, which we dont want to do

Reply:
In fact that was not a side effect I was planning on adding. I just corrected it by using a `default` empty string value.

The PR should be good to go now, but personally I think it would be better to have this method to accept an optional weaviate Client as parameter, instead of dealing with its instantiation internally. Any opinion is this other approach for the future?

Reply:
we can aboslutely do that. can make it accept both - check for client in kwargs, if exists use that, otherwise construct